pkg/pipeline/service.go

```diff
@@ -25,6 +25,15 @@ import (
 	"github.com/conduitio/conduit/pkg/foundation/metrics/measure"
 )
 
+type FailureEvent struct {
+	// ID is the ID of the pipeline which failed.
+	ID            string
+	ProvisionType ProvisionType
+	Cause         error
+}
+
+type FailureHandler func(FailureEvent)
```

Comment:
I like the handler pattern, it keeps the logic for what to do on failures outside of the pipeline service 👍

Reply:
Yeah, let's let the pipeline service's boss decide.:D